modules/puppet_proxy/puppet_class.rb

```diff
@@ -37,11 +37,25 @@ def name
   end
 
   attr_reader :params
-
-  private
   attr_reader :klass
 
   def has_module?(klass)
     !!klass.index("::")
   end
+
+  def to_json(*a)
+    {
+        'json_class' => self.class.name,
```

Comment:
Not entirely sure why this wasn't working under json 1.5.5. Switched to explicitly instantiating `Proxy::Puppet::PuppetClass` objects instead.